Documentation/Design/Lambdas/Lambda Declare.md

```diff
@@ -0,0 +1,162 @@
+# Lambda Declare
+
+## Overview
+GLS will support declaring lambda functions. In order to declare a function that takes a lambda function as a parameter, the lambda must first be declared.
```

Comment:
For Python programmers who don't know why we're doing this silly thing, can you explain?